msg-transport/src/lib.rs

```diff
@@ -58,15 +94,67 @@ impl Tcp {
     }
 }
 
+pub struct AuthLayer {
+    id: Bytes,
+}
+
+impl Layer for AuthLayer {
+    type Io = TcpStream;
```

Comment:
Why isn't AuthLayer generic over the transport type?

Reply:
You're right, it's much cleaner, I fixed that in the stack above this one!